homeassistant/components/solax/sensor.py

```diff
@@ -25,44 +26,52 @@
 SCAN_INTERVAL = timedelta(seconds=30)
 
 
+DEVICE_CLASS_MAPPING = {
+    Units.C: SensorDeviceClass.TEMPERATURE,
```

Comment:
Please define a map of entity descriptions instead. Probably use both unit of measurement and `is_monotonic` in a tuple as keys.

Example:
https://github.com/home-assistant/core/blob/051974304a7c663386bf0a5b4d1c0730df9fa21e/homeassistant/components/xiaomi_ble/sensor.py#L37-L43

Reply:
Will do.

Out of curiosity, do you have a link to discussion for why this is preferred? Or maybe you could share the reason?

Reply:
I.e. do you care about using `SensorEntityDescription`, or do you care more about mapping all permutations in a dict?

Reply:
It's easier to read and extend declarative code.